### PR TITLE
Draft for a blog post on optimization

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

File: content/2016-03-09-resample.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+.. code:: sh
+
+    % python -m timeit 'import numpy as np; np.random.seed(0) ; n = 1000; xs = np.arange(n, dtype=np.float64); qs = np.array([1.0/n,]*n); rands = np.random.rand(n); from resample import resample' 'resample(xs, qs, rands)'
+    1000 loops, best of 3: 1.23 msec per loop
+
+Second step: Pythran + OpenMP
+=============================
+
+But could we do better? An astute reader would note that the for loop can be
```
Comment (pbrunet) [inline content/2016-03-09-resample.rst]: astuce

File: content/2016-03-09-resample.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                return i
+        raise ValueError("Value not found")
+
+    def resample(qs, xs, rands):
+        results = np.empty_like(qs)
+        lookup = np.cumsum(qs)
+        for j, key in enumerate(rands):
+            i = np_index(lookup > key, True)
+            results[j] = xs[i]
+        return results
```
Comment (pbrunet) [inline content/2016-03-09-resample.rst:142]: You didn't implement an overload of argmax for fexpr? :)

File: content/2016-03-09-resample.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    #. Pythran supports built-in exceptions ;-)
+
+And a last benchmark:
+
+.. code:: sh
+
+    % pythran resample.py
+    % python -m timeit 'import numpy as np; np.random.seed(0) ; n = 1000; xs = np.arange(n, dtype=np.float64); qs = np.array([1.0/n,]*n); rands = np.random.rand(n); from resample import resample' 'resample(xs, qs, rands)'
+    1000 loops, best of 3: 64.5 usec per loop
```
Comment (pbrunet) [inline content/2016-03-09-resample.rst]: No tests with SIMD? :-)

File: content/2016-03-09-resample.rst (hunk 2 above)
Comment (serge-sans-paille) [inline content/2016-03-09-resample.rst:142]: > You didn't implement an overload of argmax for fexpr? :)

 The problem is not fexpr, but boolean. In fact for any type, if one
 value is the max possible value for this type then we could short
 circuit. It's trivial to implement and we are 100% sure it's benefical
 for bool. for other types, it adds an extra check and may not be
 profitable. I'll open an issue for this.

File: content/2016-03-09-resample.rst (hunk 3 above)
Comment (serge-sans-paille) [inline content/2016-03-09-resample.rst]: > No tests with SIMD? :-)

nothing to gain here (simd and boolean expression don't work yet)

**Pull request merged**